addons/account/models/account.py

```diff
@@ -311,8 +311,9 @@ def _compute_account_root(self):
         # this computes the first 2 digits of the account.
         # This field should have been a char, but the aim is to use it in a side panel view with hierarchy, and it's only supported by many2one fields so far.
         # So instead, we make it a many2one to a psql view with what we need as records.
-        for record in self:
-            record.root_id = (ord(record.code[0]) * 1000 + ord(record.code[1:2] or ' ')) if record.code else False
+        for record in self.filtered(lambda a: a.code):
+            record.root_id = (ord(record.code[0]) * 1000 + ord(record.code[1])) if len(record.code) > 1\
+                else ord(record.code[0]) * 1000
```

Comment:
I don't remember why I set a `... if record.code else False` because `code` is supposed to be required (maybe a weird ORM thingy), but the way you use filtered is wrong in general because it wont update the value for the record if `code` goes from something to a falsy value.
Also, what you do is the same as what I did with `ord(record.code[1:2] or ' ')`, isn't it? I guess it isn't finding it in the table because the init of `account.root` doesn't create the space.
So I think the better fix would be:
```diff
---            record.root_id = (ord(record.code[0]) * 1000 + ord(record.code[1:2] or ' ')) if record.code else False
+++            record.root_id = (ord(record.code[0]) * 1000 + ord(record.code[1:2])) if record.code else False
```
It might not be obvious but it works because
```python
"s"[1:2] == ''
```

Reply:
Do you want me to apply for this FIX and update PR? (or Wait for anyone's review)

Reply:
I didn't test it myself, but if you agree with what I proposed it would be great if you could do it

Reply:
I will check this FIX and update it here. (Probably tomorrow )
Thanks for Review

Reply:
@william-andre   It won't work with Single Char 
 `TypeError: ord() expected a character, but string of length 0 found`
```
>>> ord()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: ord() takes exactly one argument (0 given)
>>> ord('')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: ord() expected a character, but string of length 0 found
>>> ord(False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: ord() expected string of length 1, but bool found
>>> ord(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: ord() expected string of length 1, but NoneType found

```
-----------

> Also, what you do is the same as what I did with ord(record.code[1:2] or ' '), isn't it? 

Nope, My patch Generates root with either two (if len>1) or one char

Reply:
Dirty but should work well :no_good_man: 

`record.root_id = (ord(record.code[0]) * 1000 + ord(record.code[1:2] or '\x00')) if record.code else False
`

Reply:
I like dirty :+1:

Reply:
@william-andre Updated